sqlmesh/core/plan/builder.py

```diff
@@ -583,38 +584,35 @@ def _categorize_snapshots(
             if not snapshot or not self._is_new_snapshot(snapshot):
                 continue
 
+            forward_only = self._is_forward_only_change(s_id) or self._forward_only
+
             if s_id in self._choices:
-                snapshot.categorize_as(self._choices[s_id])
+                snapshot.categorize_as(self._choices[s_id], forward_only)
                 continue
 
             if s_id in self._context_diff.added:
-                snapshot.categorize_as(SnapshotChangeCategory.BREAKING)
-            elif self._is_forward_only_change(s_id) or self._forward_only:
-                # In case of the forward only plan any modifications result in reuse of the
-                # previous version for non-seed models.
-                # New snapshots of seed models are considered non-breaking ones.
-                category = (
-                    SnapshotChangeCategory.NON_BREAKING
-                    if snapshot.is_seed
-                    else SnapshotChangeCategory.FORWARD_ONLY
-                )
-                # If the model kind changes mark as breaking
-                if snapshot.is_model and snapshot.name in self._context_diff.modified_snapshots:
-                    _, old = self._context_diff.modified_snapshots[snapshot.name]
-                    if _is_breaking_kind_change(old, snapshot):
-                        category = SnapshotChangeCategory.BREAKING
-
-                snapshot.categorize_as(category)
+                snapshot.categorize_as(SnapshotChangeCategory.BREAKING, forward_only)
             elif s_id.name in self._context_diff.modified_snapshots:
-                self._categorize_snapshot(snapshot, dag, indirectly_modified)
+                self._categorize_snapshot(snapshot, forward_only, dag, indirectly_modified)
 
     def _categorize_snapshot(
-        self, snapshot: Snapshot, dag: DAG[SnapshotId], indirectly_modified: SnapshotMapping
+        self,
+        snapshot: Snapshot,
+        forward_only: bool,
+        dag: DAG[SnapshotId],
+        indirectly_modified: SnapshotMapping,
     ) -> None:
         s_id = snapshot.snapshot_id
 
         if self._context_diff.directly_modified(s_id.name):
-            if self._auto_categorization_enabled:
+            new, old = self._context_diff.modified_snapshots[s_id.name]
+            if _is_breaking_kind_change(old, new):
+                snapshot.categorize_as(SnapshotChangeCategory.BREAKING, False)
+            elif self._auto_categorization_enabled:
```

Comment:
Why can you check if `_is_breaking_kind_change` and categorize even if auto categorization is not enabled?

Reply:
because the kind change is unconditionally breaking? Is this too aggressive?

Reply:
If someone didn't enable auto-categorization then they want to be in control of the change category and not defer to us even if we think it should be breaking. Or am I not understanding this part? I might not be understanding the full nuance you intend with "unconditionally breaking".